test/e2e/suite/test_node.go

```diff
@@ -172,15 +220,20 @@ func (n *testNode) It(name string, f func()) {
 	n.serialOutput.It(name, f)
 }
 
-func (n *testNode) PeerdNode() *peered.PeerdNode {
-	return n.node
+func (n *testNode) PeeredInstance() *peered.PeeredInstance {
+	return n.peeredInstance
 }
 
 func (n *testNode) isImpaired(ctx context.Context, waitErr error) bool {
 	if waitErr == nil {
 		return false
 	}
-	isImpaired, err := ec2.IsEC2InstanceImpaired(ctx, n.EC2Client, n.node.Instance.ID)
+	isImpaired, err := ec2.IsEC2InstanceImpaired(ctx, n.EC2Client, n.peeredInstance.ID)
 	n.Logger.Error(err, "describing instance status")
 	return isImpaired
 }
+
+// SetNodeWaiter allows setting a custom NodeWaiter for the testNode
+func (n *testNode) SetNodeWaiter(waiter NodeWaiter) {
```

Comment:
this is a bit weird
So we add new arguments to the constructor for some of the things we want to make configurable
But then for this one we create a setter. Why the difference?

Looking at this, it makes me think that some kind of optional options in the constructor is probably the way to go. that way you can configure this or other fields for bottlerocket but let the other usecases take advantage of a default.

Reply:
Done